packages/context/src/inject.ts

```diff
@@ -112,6 +117,9 @@ export function inject(
     resolve = resolveValuesByFilter;
   }
   const injectionMetadata = Object.assign({decorator: '@inject'}, metadata);
+  if (injectionMetadata.bindingSorter && !resolve) {
+    throw new Error('Binding sorter is only allowed with a binding filter');
```

Comment:
For my understanding, what's the rationale behind not allowing binding sorting without binding filter?

Reply:
When there is no binding filter, then the injection is resolved with a single binding (as described by the provided binding key). It does not make sense to sort a single binding.

At least that's my understanding.